### PR TITLE
Add files for lesson 5

### PR DESCRIPTION
**Pull request by Resires**
Some problems are added with "test_*" file.
This time problems are splited per folders according to following order:
1 Automate the Boring Stuff with Python / Chapter 4 / Comma Code
2 Automate the Boring Stuff with Python / Chapter 4 / Character Picture Grid
3 Snakify / Lesson 7 / Problems ("Greater than neighbours", "Swap min and max", "The bowling alley")
4 Snakify / Lesson 9 / Problems ("Maximum", "Chess board", "Swap the columns" (optional), "The diagonal parallel to the main")
5 Numbers filter
6 Age calculator

### CODE REVIEW

File: students/uss_tomasz/lesson_05_lists/5_1/comma_code.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+def list_to_string(list_from_user):
+    """ This function takes a list value as an argument and returns a string with all the items separated by a comma
+    and a space, with word 'and' insterted before the last item. Example:
+    Input = ['apples', 'bananas', 'tofu', 'cats']
+    Output = 'apples, bananas, tofu, and cats'
+    """
+    string_for_user = ''
+    for index, item in enumerate(list_from_user):
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_1/comma_code.py]: `enumerate`, very good

File: students/uss_tomasz/lesson_05_lists/5_1/comma_code.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+def list_to_string(list_from_user):
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_1/comma_code.py]: in fact this function is should not care if the list is from user or not, so the name `list_from_user` is not the best choice

File: students/uss_tomasz/lesson_05_lists/5_1/test_comma_code.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    def test_integers(self):
+        self.assertEqual(list_to_string([1, 3]), '1 and 3')
+
+    def test_none_arguments(self):
+        self.assertEqual(list_to_string([None, None]), 'None and None')
+
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_1/test_comma_code.py:26]: 🥇

File: students/uss_tomasz/lesson_05_lists/5_2/character_picture_grid.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+def make_grid_transpose(grid_from_user):
+    column_quantity = len(grid_from_user[0])
+    return [[row[i] for row in grid_from_user] for i in range(column_quantity)]
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_2/character_picture_grid.py:3]: looks compact and clever, but it's hard to follow what's going on here
for production code it should rather be changed to more verbose solution, but here we can keep it as it is

File: students/uss_tomasz/lesson_05_lists/5_3/bowling_alley.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+from random import randint
+pins_quantity = 10
+balls_quantity = 5
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_3/bowling_alley.py]: these 2 are constants and should be named according to PEP8 https://www.python.org/dev/peps/pep-0008/#constants

File: students/uss_tomasz/lesson_05_lists/5_3/bowling_alley.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+pins = ['I']*pins_quantity
+
+
+def throw_ball():
+    """Returns randomly positions range. Between them pins will be removed from bowling alley.
+    First value is always smaller than a second value
+    """
+    a, b = randint(1, 10), randint(1, 10)
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_3/bowling_alley.py]: **Optional**
Strive for clarity, not for minimal number of lines.
This expression introduces unnecessary binding between these 2 variables, where in fact there is no correlation between them:
```
a, b = randint(1, 10), randint(1, 10)
```
This expression on the other hand clearly shows that they are separate
```
a = randint(1, 10)
b = randint(1, 10)
```

File: students/uss_tomasz/lesson_05_lists/5_3/bowling_alley.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+print("Initial pins in the bowling alley =", pins)
+for i in range(balls_quantity):
+    start_position, stop_position = throw_ball()
+    if start_position == stop_position:
+        print("Your ball hit pin from", start_position, 'position!')
+    else:
+        print("Your ball hit pins from", start_position, "up to", stop_position, "position!")
+    pins = pins[:start_position-1] + ['.']*(stop_position-start_position+1) + pins[stop_position:]
+    print("Pins remained: ", pins)
+    if pins == ['.'] * 10:
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_3/bowling_alley.py]: **optional**
it's more performant to define 
```
PINS_ALL_DOWN = ['.'] * 10
```
 upfront, not calculate on every loop. It doesn't matter here, but for larger arrays it would

File: students/uss_tomasz/lesson_05_lists/5_3/snakify_lesson_7_problems.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+def quantity_of_bigger_than_both_neighbours(list_from_user):
+    quantity = 0
+    for i in range(1, len(list_from_user) - 1):
+        if list_from_user[i] > list_from_user[i-1] and list_from_user[i] > list_from_user[i+1]:
+            quantity += 1
+    return quantity
+
+
+list_for_test = remove_duplicates([randint(1, 10) for i in range(10)])
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_3/snakify_lesson_7_problems.py:19]: **Super important**
Tests must be always run with the same deterministic data set. Otherwise you'll end up trying to reproduce bugs that appear in your tests once in 20 runs or worse. This is productivity killer.

Always seed your random number generator before your you it in tests.

File: students/uss_tomasz/lesson_05_lists/5_4/Snakify_Lesson_9_Problems.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
+# https://snakify.org/lessons/two_dimensional_lists_arrays/problems/
+from random import randint
+# Problem Maximum
+rows_quantity = 6
+columns_quantity = 4
+array = [[randint(1, 30) for i in range(columns_quantity)] for j in range(rows_quantity)]
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_4/Snakify_Lesson_9_Problems.py]: same as above

File: students/uss_tomasz/lesson_05_lists/5_4/Snakify_Lesson_9_Problems.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+def print_matrix(matrix_from_user):
+    for row in matrix_from_user:
+        for element in row:
+            print(element, end=' ')
+        print()
+
+
+rows_quantity = 4
+columns_quantity = 6
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_4/Snakify_Lesson_9_Problems.py]: name your consts according to PEP8

File: students/uss_tomasz/lesson_05_lists/5_5/numbers_filter.py [hunk 11]
```diff
@@ -0,0 +1,22 @@
+# Problem - Numbers filter
+def strings_list_to_integers_list(strings_list_from_user, filter_range):
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_5/numbers_filter.py]: Same as above - this function should not care if the list is from user or any other source.

File: students/uss_tomasz/lesson_05_lists/5_5/numbers_filter.py [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+    to_filter_range = range(3)
+    expected_output = [8, 3]
+    """
+    integers_list_for_user = []
+    for element in strings_list_from_user:
+        integers_list_for_user.append(int(element))
+    for i in filter_range:
+        while i in integers_list_for_user:      # The same item can exist multiple times
+            integers_list_for_user.remove(i)
+    return integers_list_for_user
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_5/numbers_filter.py]: Although this solution works, you didn't use list comprehension and its filtering capabilities here. Please refactor.

File: students/uss_tomasz/lesson_05_lists/5_6/age_calculator.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+    average age of adults (age >= 18)
+    count the children (age < 18)
+    """
+    adults_list = [i for i in peoples_age_list if i >= 18]
+    average_adult = sum(adults_list)/len(adults_list)
+    children_count = len([i for i in peoples_age_list if i < 18])
+    return average_adult, children_count
+
+
+group_of_people = [randint(1, 30) for i in range(10)]
```
Comment (jedzej) [inline students/uss_tomasz/lesson_05_lists/5_6/age_calculator.py:17]: same as above

**Review by jedzej — CHANGES_REQUESTED** [review]
Some comments in code.

Please read about these libraries
https://github.com/joke2k/faker
https://factoryboy.readthedocs.io/en/latest/
They are good examples of libraries used for providing random, yet predictable data sets for some more complex testing tasks.

File: students/uss_tomasz/lesson_05_lists/5_6/age_calculator.py (hunk 13 above)
Comment (Resires) [reply students/uss_tomasz/lesson_05_lists/5_6/age_calculator.py:17]: Please double-check. List comprahensions are used in lines 10 and 12. Or your comment was related with other issue?

Comment (jedzej) [reply students/uss_tomasz/lesson_05_lists/5_6/age_calculator.py:17]: It refers to unseeded random number generator.

**Review by jedzej — APPROVED** [review]
well done
